core/web/router.go

```diff
@@ -333,22 +333,24 @@ func guiAssetRoutes(box packr.Box, engine *gin.Engine) {
 			}
 		}
 
-		if matchedBoxPath != "" {
-			file, err := box.Open(matchedBoxPath)
-			if err != nil {
-				if err == os.ErrNotExist {
-					c.AbortWithStatus(http.StatusNotFound)
-				} else {
-					err := fmt.Errorf("failed to open static file '%s': %+v", path, err)
-					logger.Error(err.Error())
-					jsonAPIError(c, http.StatusInternalServerError, err)
-				}
-				return
-			}
-			defer file.Close()
+		if matchedBoxPath == "" {
+			matchedBoxPath = "404.html"
+		}
 
-			http.ServeContent(c.Writer, c.Request, path, time.Time{}, file)
+		file, err := box.Open(matchedBoxPath)
+		if err != nil {
+			if err == os.ErrNotExist {
+				c.AbortWithStatus(http.StatusNotFound)
+			} else {
+				err := fmt.Errorf("failed to open static file '%s': %+v", path, err)
+				logger.Error(err.Error())
+				jsonAPIError(c, http.StatusInternalServerError, err)
```

Comment:
@spooktheducks ❤️ the change in general but I feel like this shouldn't return an error in JSON-API format as this will be returning static assets like `html`, `js`, `css`. A standard http internal server error should suffice.

Reply:
:+1: Good catch.